andela/firestar-backend
# #167730740 Build Email Verification feature on user sign up

#### What does this PR do?
It Build Email Verification feature on user sign up and sends email for verification to user on successful sign up
### Description of Task to be completed?
N/A
#### How should this be manually tested?

1. ```git clone -b ft-email-verification-167730740  https://github.com/andela/firestar-backend.git .```  @[ft-email-verification-167730740](https://github.com/andela/firestar-backend/tree/ft-email-verification-167730740)
2. run `npm install` and create  a `.env file` and provide the following information to it 
```
EMAIL = 
LAYER=
PASSWORD=
SECRET_KEY_EMAIL_VERIFY_TOKEN =
PORT = 
```
 
4. start the server with `npm run start:dev`

5. Now use postman and go to `localhost:PORT/api/v1/users/email/test` using a `POST` request with a body of 
```json
body:{ 
email: 'youremail',
firstName: 'Yourfirstname'
lastName: 'Yourlastname'
}
```
6. If successful you would see a response similar to the response of the screenshot on postman
7. Visit the email you provided and see the newly delivered email which contains your name and link to verify your account.
8. copy to clipboard the `id key` token from the url sent to the provided mail and use it on postman on a `get` request on `http://localhost:PORT/api/v1/users/email/verify?id=token` to verify the token sent

#### Any background context you want to provide?
Please note that a token expires after 24hrs
#### What are the relevant pivotal tracker stories?
[#167730740](https://www.pivotaltracker.com/story/show/167730740)

#### Screenshots or gifs (if appropriate)
**Test coverage information**

![image](https://user-images.githubusercontent.com/38135488/63853686-40707c80-c993-11e9-93fc-c91e179662d4.png)

* 
![Screenshot (29)](https://user-images.githubusercontent.com/38135488/63482714-cec79880-c491-11e9-988b-cb71f17459a9.png)

![image](https://user-images.githubusercontent.com/38135488/63848879-a73c6880-c988-11e9-8449-0b64819c6bc1.png)



#### Questions:
* N/A
